README.md

```diff
@@ -0,0 +1,101 @@
+# Node.js Version Installer
+
+A command-line tool to make Node.js binaries available in CI environments.
+
+## Can't I just use nvm?
+
+We've used `nvm` for local Node.js development for a long time. It works very
+well on your local machine, where you tend to have long-running bash sessions
+and tend to frequently switch between Node.js versions. However, it has not been
+as painless to use in CI environments, where your bash sessions tend to be very
+brief and interspersed across machines, docker containers, and build jobs.
+
+What causes this friction between CI environments and `nvm`? The answer is
+essentially: bash functions.
+
+The way `nvm` makes its functionality available to the user is as a bash
+function by the same name. This bash function is loaded using your `~/.bashrc`
+or `~/.bash_profile` file, and it lives while your bash session does. In
+particular, `nvm` is not an executable file, it is not on your `$PATH`, and it
+is not a system-wide command.
+
+The way most recent CI environments work is by letting you group several
+statements into a "job", "stage", or something similar, and then execute these
+jobs in fresh Docker containers based on a small set of common images. In this
+setup everything is running in its own containerized little world, and there is
+no guarantee that each step in a CI job is executed within a bash context. In
+our current CI, there is no such bash context, no `.bashrc` or `.bash_profile`
+is loaded, and it generally requires additional workarounds to make this happen.
+
+## Another approach: `nvi`
+
+We really want to use containerized CI, so we considered how other dependencies
+are injected in our Docker containers, and how to do the same for our Node.js
+version installer. A common solution is to download an executable, put it on the
+path, and then build it into the image used by your CI. This is exactly what
+`nvi` strives to support.
+
+### Design criteria
+
+`nvi` strives to:
+
+0.  Be a simple way to install Node.js versions in CI environments.
+0.  Let you install specific versions of executables for `node` and `npm`.
+0.  Let you place these `node` and `npm` executables on your `$PATH`.
+0.  Let you specify your desired Node.js version in package.json.
+
+`nvi` avoids:
+
+0.  Relying on bash functions.
```

Comment:
This is a pretty poor phrasing. I'm hoping to convey that _as a user_ you don't have to put up with bash function nonsense. The internal implementation should be free to use whatever means it want, like bash functions.